### ISSUE TITLE
Mariadb-init is not getting completed

### ISSUE THREAD

**Issue by Helloworld1707** [stackanetes/stackanetes#152]
Hi,
I'm trying to install stackanetes , and while deploying I noticed that the mariadb-init-xxxx is struck while initializing, So that the actual 3 instances of DB containers "mariadb-xxxx" are not getting created. 
please help ... 

```
********************************************************************************
 kubectl get pods --namespace=openstack | grep maria
mariadb-init-47vww                  1/1       Running            0          19h

*******************************************************************************
kubectl logs mariadb-init-47vww --namespace=openstack
Entrypoint INFO: 2017/02/22 02:00:11 entrypoint.go:47: Resolving /etc/my.cnf.d/wsrep.cnf
Entrypoint INFO: 2017/02/22 02:00:11 config.go:67: /etc/my.cnf.d/wsrep.cnf
Entrypoint INFO: 2017/02/22 02:00:13 entrypoint.go:56: Dependency /etc/my.cnf.d/wsrep.cnf is resolved
+ SLEEP_TIMEOUT=5
++ python /tmp/replicas.py
+ REPLICAS=0
+ '[' 0 -eq 1 ']'
+ '[' 0 -eq 2 ']'
+ bash /tmp/bootstrap-db.sh
+ SLEEP_TIMEOUT=5
+ mysql_install_db --datadir=/var/lib/mysql
Installing MariaDB/MySQL system tables in '/var/lib/mysql' ...
170222  2:00:13 [Warning] options --log-slow-admin-statements, --log-queries-not-using-indexes and --log-slow-slave-statements have no effect if --log_slow_queries is not set
170222  2:00:13 [Note] /usr/sbin/mysqld (mysqld 10.0.27-MariaDB-1~jessie-wsrep) starting as process 54 ...
/usr/sbin/mysqld: Query cache is disabled (resize or similar command in progress); repeat this command later
170222  2:00:13 [Note] WSREP: Read nil XID from storage engines, skipping position init
170222  2:00:13 [Note] WSREP: wsrep_load(): loading provider library 'none'
2017-02-22 02:00:13 7fbc7248d780 InnoDB: Warning: Using innodb_locks_unsafe_for_binlog is DEPRECATED. This option may be removed in future releases. Please use READ COMMITTED transaction isolation level instead, see http://dev.mysql.com/doc/refman/5.6/en/set-transaction.html.
170222  2:00:13 [Note] InnoDB: Using mutexes to ref count buffer pool pages
170222  2:00:13 [Note] InnoDB: The InnoDB memory heap is disabled
170222  2:00:13 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
170222  2:00:13 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
170222  2:00:13 [Note] InnoDB: Compressed tables use zlib 1.2.8
170222  2:00:13 [Note] InnoDB: Using Linux native AIO
170222  2:00:13 [Note] InnoDB: Using CPU crc32 instructions
170222  2:00:13 [Note] InnoDB: Initializing buffer pool, size = 256.0M
170222  2:00:13 [Note] InnoDB: Completed initialization of buffer pool
170222  2:00:13 [Note] InnoDB: The first specified data file ./ibdata1 did not exist: a new database to be created!
170222  2:00:13 [Note] InnoDB: Setting file ./ibdata1 size to 12 MB
170222  2:00:13 [Note] InnoDB: Database physically writes the file full: wait...
170222  2:00:13 [Note] InnoDB: Setting log file ./ib_logfile101 size to 2000 MB
InnoDB: Progress in MB: 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000
170222  2:00:25 [Note] InnoDB: Setting log file ./ib_logfile1 size to 2000 MB
InnoDB: Progress in MB: 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000
170222  2:00:37 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
170222  2:00:37 [Warning] InnoDB: New log files created, LSN=45781
170222  2:00:37 [Note] InnoDB: Doublewrite buffer not found: creating new
170222  2:00:37 [Note] InnoDB: Doublewrite buffer created
170222  2:00:37 [Note] InnoDB: 128 rollback segment(s) are active.
170222  2:00:37 [Warning] InnoDB: Creating foreign key constraint system tables.
170222  2:00:37 [Note] InnoDB: Foreign key constraint system tables created
170222  2:00:37 [Note] InnoDB: Creating tablespace and datafile system tables.
170222  2:00:37 [Note] InnoDB: Tablespace and datafile system tables created.
170222  2:00:37 [Note] InnoDB: Waiting for purge to start
170222  2:00:37 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.31-77.0 started; log sequence number 0
170222  2:00:38 [Warning] Failed to load slave replication state from table mysql.gtid_slave_pos: 1146: Table 'mysql.gtid_slave_pos' doesn't exist
170222  2:00:40 [Note] WSREP: Service disconnected.
170222  2:00:41 [Note] WSREP: Some threads may fail to exit.
170222  2:00:41 [Note] InnoDB: FTS optimize thread exiting.
170222  2:00:41 [Note] InnoDB: Starting shutdown...
170222  2:00:43 [Note] InnoDB: Shutdown completed; log sequence number 1616697
OK
Filling help tables...
170222  2:00:43 [Warning] options --log-slow-admin-statements, --log-queries-not-using-indexes and --log-slow-slave-statements have no effect if --log_slow_queries is not set
170222  2:00:43 [Note] /usr/sbin/mysqld (mysqld 10.0.27-MariaDB-1~jessie-wsrep) starting as process 89 ...
/usr/sbin/mysqld: Query cache is disabled (resize or similar command in progress); repeat this command later
170222  2:00:43 [Note] WSREP: Read nil XID from storage engines, skipping position init
170222  2:00:43 [Note] WSREP: wsrep_load(): loading provider library 'none'
2017-02-22 02:00:43 7fa8758e4780 InnoDB: Warning: Using innodb_locks_unsafe_for_binlog is DEPRECATED. This option may be removed in future releases. Please use READ COMMITTED transaction isolation level instead, see http://dev.mysql.com/doc/refman/5.6/en/set-transaction.html.
170222  2:00:43 [Note] InnoDB: Using mutexes to ref count buffer pool pages
170222  2:00:43 [Note] InnoDB: The InnoDB memory heap is disabled
170222  2:00:43 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
170222  2:00:43 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
170222  2:00:43 [Note] InnoDB: Compressed tables use zlib 1.2.8
170222  2:00:43 [Note] InnoDB: Using Linux native AIO
170222  2:00:43 [Note] InnoDB: Using CPU crc32 instructions
170222  2:00:43 [Note] InnoDB: Initializing buffer pool, size = 256.0M
170222  2:00:43 [Note] InnoDB: Completed initialization of buffer pool
170222  2:00:43 [Note] InnoDB: Highest supported file format is Barracuda.
170222  2:00:43 [Note] InnoDB: 128 rollback segment(s) are active.
170222  2:00:43 [Note] InnoDB: Waiting for purge to start
170222  2:00:43 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.31-77.0 started; log sequence number 1616697
170222  2:00:43 [Note] WSREP: Service disconnected.
170222  2:00:44 [Note] WSREP: Some threads may fail to exit.
170222  2:00:44 [Note] InnoDB: FTS optimize thread exiting.
170222  2:00:44 [Note] InnoDB: Starting shutdown...
170222  2:00:46 [Note] InnoDB: Shutdown completed; log sequence number 1616707
OK

To start mysqld at boot time you have to copy
support-files/mysql.server to the right place for your system

PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
To do so, start the server, then issue the following commands:

'/usr/bin/mysqladmin' -u root password 'new-password'
'/usr/bin/mysqladmin' -u root -h mariadb-init-47vww password 'new-password'

Alternatively you can run:
'/usr/bin/mysql_secure_installation'

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the MariaDB Knowledgebase at http://mariadb.com/kb or the
MySQL manual for more instructions.

You can start the MariaDB daemon with:
cd '/usr' ; /usr/bin/mysqld_safe --datadir='/var/lib/mysql'

You can test the MariaDB daemon with mysql-test-run.pl
cd '/usr/mysql-test' ; perl mysql-test-run.pl

Please report any problems at http://mariadb.org/jira

The latest information about MariaDB is available at http://mariadb.org/.
You can find additional information about the MySQL part at:
http://dev.mysql.com
Support MariaDB development by buying support/new features from MariaDB
Corporation Ab. You can contact us about this at sales@mariadb.com.
Alternatively consider joining our community based development effort:
http://mariadb.com/kb/en/contributing-to-the-mariadb-project/

+ TIMEOUT=120
+ mysqld_safe --defaults-file=/etc/my.cnf --console --wsrep-new-cluster --wsrep_cluster_address=gcomm:// --bind-address=127.0.0.1 --wsrep_node_address=127.0.0.1 '--wsrep_provider_options=gcache.size=512M; gmcast.listen_addr=tcp://127.0.0.1:4567'
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 120 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
170222 02:00:47 mysqld_safe Logging to '/var/lib/mysql/mariadb-init-47vww.err'.
170222 02:00:47 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
170222 02:00:47 mysqld_safe WSREP: Running position recovery with --log_error='/var/lib/mysql/wsrep_recovery.6DZrE6' --pid-file='/var/lib/mysql/mariadb-init-47vww-recover.pid'
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 119 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 118 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 117 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
170222 02:00:50 mysqld_safe WSREP: Recovered position 00000000-0000-0000-0000-000000000000:-1
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ cd /var/lib/mysql
+ DB_ROOT_PASSWORD=password
+ kolla_security_reset
spawn mysql_secure_installation

NOTE: RUNNING ALL PARTS OF THIS SCRIPT IS RECOMMENDED FOR ALL MariaDB
      SERVERS IN PRODUCTION USE!  PLEASE READ EACH STEP CAREFULLY!

In order to log into MariaDB to secure it, we'll need the current
password for the root user.  If you've just installed MariaDB, and
you haven't set the root password yet, the password will be blank,
so you should just press enter here.

Enter current password for root (enter for none):
OK, successfully used password, moving on...

Setting the root password ensures that nobody can log into the MariaDB
root user without the proper authorisation.

Set root password? [Y/n] y
New password:
Re-enter new password:
Password updated successfully!
Reloading privilege tables..
 ... Success!


By default, a MariaDB installation has an anonymous user, allowing anyone
to log into MariaDB without having to have a user account created for
them.  This is intended only for testing, and to make the installation
go a bit smoother.  You should remove them before moving into a
production environment.

Remove anonymous users? [Y/n] y
 ... Success!

Normally, root should only be allowed to connect from 'localhost'.  This
ensures that someone cannot guess at the root password from the network.

Disallow root login remotely? [Y/n] n
 ... skipping.

By default, MariaDB comes with a database named 'test' that anyone can
access.  This is also intended only for testing, and should be removed
before moving into a production environment.

Remove test database and access to it? [Y/n] y
 - Dropping test database...
 ... Success!
 - Removing privileges on test database...
 ... Success!

Reloading the privilege tables will ensure that all changes made so far
will take effect immediately.

Reload privilege tables now? [Y/n] y
 ... Success!

Cleaning up...

All done!  If you've completed all of the above steps, your MariaDB
installation should now be secure.

Thanks for using MariaDB!
+ mysql -u root --password=password --port=3306 -e 'GRANT ALL PRIVILEGES ON *.* TO '\''root'\''@'\''localhost'\'' IDENTIFIED BY '\''password'\'' WITH GRANT OPTION;'
+ mysql -u root --password=password --port=3306 -e 'GRANT ALL PRIVILEGES ON *.* TO '\''root'\''@'\''%'\'' IDENTIFIED BY '\''password'\'' WITH GRANT OPTION;'
+ mysqladmin -uroot -ppassword --port=3306 shutdown
+ SHUTDOWN_TIMEOUT=60
+ [[ -f /var/lib/mysql/mariadb.pid ]]
+ [[ 60 -gt 0 ]]
+ let SHUTDOWN_TIMEOUT-=1
+ sleep 1
+ [[ -f /var/lib/mysql/mariadb.pid ]]
+ [[ 59 -gt 0 ]]
+ let SHUTDOWN_TIMEOUT-=1
+ sleep 1
+ [[ -f /var/lib/mysql/mariadb.pid ]]
+ [[ 58 -gt 0 ]]
+ let SHUTDOWN_TIMEOUT-=1
+ sleep 1
+ [[ -f /var/lib/mysql/mariadb.pid ]]
+ [[ 57 -gt 0 ]]
+ let SHUTDOWN_TIMEOUT-=1
+ sleep 1
170222 02:00:55 mysqld_safe mysqld from pid file /var/lib/mysql/mariadb.pid ended
+ [[ -f /var/lib/mysql/mariadb.pid ]]
+ wait_for_cluster
+ TIMEOUT=120
+ mysqld_safe --defaults-file=/etc/my.cnf --console --wsrep-new-cluster --wsrep_cluster_address=gcomm://
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 120 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
170222 02:00:55 mysqld_safe Logging to '/var/lib/mysql/mariadb-init-47vww.err'.
170222 02:00:55 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
170222 02:00:55 mysqld_safe WSREP: Running position recovery with --log_error='/var/lib/mysql/wsrep_recovery.bsteB5' --pid-file='/var/lib/mysql/mariadb-init-47vww-recover.pid'
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 119 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 118 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
170222 02:00:58 mysqld_safe WSREP: Recovered position bc0a0651-f8a2-11e6-bbd5-07dcacc8bab5:5
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
+ [[ 117 -gt 0 ]]
+ let TIMEOUT-=1
+ sleep 1
+ [[ ! -f /var/lib/mysql/mariadb.pid ]]
++ python /tmp/replicas.py
+ REPLICAS=0
+ MINIMUM_CLUSTER_SIZE=1
+ true
++ mysql -uroot -ppassword --port=3306 '-eshow status'
++ grep wsrep_cluster_size
++ awk ' { if($2 ~ /[0-9]/){ print $2 } else { print 0 } } '
+ CLUSTER_SIZE=1
+ '[' 1 -lt 1 ']'
++ mysql -uroot -ppassword --port=3306 '-eshow status'
++ grep wsrep_local_state_comment
++ awk ' { print $2 } '
+ CLUSTER_STATUS=Synced
+ '[' Synced '!=' Synced ']'
++ python /tmp/peer-finder.py mariadb
++ grep -o ,
++ wc -l
**********************************************************************************************
```

**Comment by PAStheLoD** [comment]
Hello,

just a quick meta-comment, please use backticks around log/code text, so it'll be easier to read (and markdown won't parse and style it), like this:

\```
code
\```

becomes

```
code
```

**Comment by ss7pro** [comment]
@Helloworld1707 It looks that you're missing appropriate labels on cluster nodes, can you show output of:

kubectl get nodes --show-labels



Cheers

**Comment by Helloworld1707** [comment]
kubectl get nodes --show-labels
NAME      STATUS                     AGE       LABELS
kube-1     Ready,SchedulingDisabled   1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-1
kube-2     Ready,SchedulingDisabled   1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-2
kube-5     Ready                      1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-5,openstack-control-plane=enabled
kube-6     Ready                      1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-6,openstack-control-plane=enabled
kube-7     Ready                      1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-7,openstack-control-plane=enabled
kube-8     Ready                      1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-8,openstack-compute-node=enabled
kube-9     Ready                      1d        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-9,openstack-compute-node=enabled
[root@kube-1 ~]#

**Comment by Helloworld1707** [comment]
can somebody help me here ?

**Comment by Helloworld1707** [comment]
Can somebody comment on 
++ python /tmp/replicas.py
+ REPLICAS=0

and is this  the reason for this problem  ?

**Comment by Quentin-M** [comment]
Hi,

There is definitely something odd here. Could you please describe the `mariadb` DaemonSet, as well as its pods? There should be 3 desired replicas described there, and therefore the `REPLICAS` should have 3 as its value.

**Comment by Helloworld1707** [comment]
Hi ,

Here is the O/P

kubectl describe  DaemonSet mariadb  --namespace=openstack
Name:           mariadb
Image(s):       quay.io/stackanetes/stackanetes-mariadb:newton
Selector:       galera=enabled
Node-Selector:  openstack-control-plane=enabled
Labels:         app=mariadb
                galera=enabled
Desired Number of Nodes Scheduled: 0
Current Number of Nodes Scheduled: 0
Number of Nodes Misscheduled: 0
Pods Status:    0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Events:
  FirstSeen     LastSeen        Count   From            SubObjectPath   Type            Reason          Message
  ---------     --------        -----   ----            -------------   --------        ------          -------
  5d            3m              1587    {daemon-set }                   Warning         FailedCreate    Error creating: pods "" is forbidden: pod.Spec.SecurityContext.RunAsUser is forbidden
[root@kube-1 ~]#

**Comment by Helloworld1707** [comment]
Hi Quentin-M,

Will you able to help me here ?

**Comment by Helloworld1707** [comment]
Hi ,
Can somebody suggest me here ?

**Comment by Quentin-M** [comment]
The issue is clear. No MariaDB instance can be started due to a policy on your Kubernetes cluster.

`Error creating: pods "" is forbidden: pod.Spec.SecurityContext.RunAsUser is forbidden`

First, you need `--allow-privileged=true` (as documented) and then, you need a lenient pod admission control policy. There is no way to run OpenStack without having a pretty wide range of permissions - how would we run VMs and have user-defined networking otherwise?

**Comment by Merlin83b** [comment]
I came across the same issue, running on a CoreOS bare metal Kubernetes installation.  Expanding on the post from @Quentin-M above and after some more Googling, I ended up removing `SecurityContextDeny` from the `/etc/kubernetes/manifests/kube-apiserver.yaml` file.  Note that this is no longer specified in the current version of the [CoreOS kubernetes installation instructions](https://coreos.com/kubernetes/docs/latest/deploy-master.html) but was present when I created the cluster.
